### PR TITLE
Create GitHub Actions workflow to release to Docker Hub

### PR DESCRIPTION
**Pull request by jillianwilson**
Uses Goreleaser to create binaries for each tagged release. Addresses https://github.com/1Password/onepassword-operator/issues/20

### CODE REVIEW

**Comment by jpcoenen** [comment]
Yay to GoReleaser :grin: ! This will save us so much time and effort in the long run.

To address the request from #20, I think we have to add a `.goreleaser.yml` that builds a Docker image for `arm64` and `amd64`. I have published a sample for this in the `goreleaser` branch. Together with a [few minor additions](https://carlosbecker.com/posts/multi-platform-docker-images-goreleaser-gh-actions/#github-actions) we should be good for automatic releases of Docker images :muscle:

**Comment by jpcoenen** [comment]
Added the steps to build and publish the images. I've not been able to test it yet. 

@jillianwilson What do you think?

**Comment by verkaufer** [comment]
Do we have plans for how we will manage that DockerHub token? @jpcoenen

**Comment by jpcoenen** [comment]
> Do we have plans for how we will manage that DockerHub token? @jpcoenen

@thatguygriff provided a token, which has been added as a secret.

**Comment by jpcoenen** [comment]
Did some local testing and this current setup with GoReleaser also pushes the images that are part of the manifest to DockerHub. So you end up with three tags: `1.0.0` (which contains both an amd64 and arm64 image), `1.0.0-amd64` and `1.0.0-arm64`. GoReleaser requires pushing these last two tags, which makes the DockerHub repo kind of a mess.

So I'll rewrite it to just use the `buildx` command. Makes things easier anyway. Then we only have 1 entry on DockerHub per version.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  goreleaser:
```
Comment (florisvdg) [inline .github/workflows/release.yml:9]: Thought that this PR was about releasing binaries as GitHub release assets _and_ pushing the Docker images.
Do we still want to do that or only do the latter?

For the latter, we're not using `goreleaser` anymore:
```suggestion
  release-docker:
```

Comment (jpcoenen) [reply .github/workflows/release.yml:9]: I interpreted the original intention to be addressing #20, which is specifically a request for building an arm64 image, not binary.

We can also publish the binaries, but do we currently have an application for them?

Comment (florisvdg) [reply .github/workflows/release.yml:9]: Not really, so I'd say let's leave that out for now.

**Comment by florisvdg** [comment]
Would be good to test this. Either on a fork or using [nektos/act](https://github.com/nektos/act).

**Comment by jpcoenen** [comment]
> Would be good to test this. Either on a fork or using [nektos/act](https://github.com/nektos/act).

Tried that, but ran into some issues with running `act` on a tag. Just figured out that apparently you can just create a local tag and run act. 

Published images to my personal DockerHub account and all seems to work nicely. 

Ran a test for `v1.1.0` and it resulted in the following three tags:
image

All of them have both an `arm64` and `amd64` image.

**Review by florisvdg — APPROVED** [review]

**Pull request merged**